File: src/test/java/in/dragonbra/javasteam/steam/handlers/steamcloud/AppFileInfoTest.java
package in.dragonbra.javasteam.steam.handlers.steamcloud;

import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesCloudSteamclient;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class AppFileInfoTest {

    @Test
    public void hasPathPrefixIndexIsFalseWhenFieldIsMissing() {
        var response = SteammessagesCloudSteamclient.CCloud_AppFileInfo.newBuilder()
                .setFileName("save.dat")
                .build();

        var info = new AppFileInfo(response);

        Assertions.assertFalse(info.getHasPathPrefixIndex());
    }

    @Test
    public void hasPathPrefixIndexIsTrueWhenIndexIsZero() {
        var response = SteammessagesCloudSteamclient.CCloud_AppFileInfo.newBuilder()
                .setFileName("save.dat")
                .setPathPrefixIndex(0)
                .build();

        var info = new AppFileInfo(response);

        Assertions.assertTrue(info.getHasPathPrefixIndex());
        Assertions.assertEquals(0, info.getPathPrefixIndex());
    }
}

File: src/test/java/in/dragonbra/javasteam/steam/handlers/steamcloud/AppFileInfoTest.kt
package `in`.dragonbra.javasteam.steam.handlers.steamcloud

import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesCloudSteamclient.CCloud_AppFileInfo
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class AppFileInfoTest {

    @Test
    fun hasPathPrefixIndexIsFalseWhenFieldIsMissing() {
        val response = CCloud_AppFileInfo.newBuilder()
            .setFileName("save.dat")
            .build()

        val info = AppFileInfo(response)

        assertFalse(info.hasPathPrefixIndex)
    }

    @Test
    fun hasPathPrefixIndexIsTrueWhenIndexIsZero() {
        val response = CCloud_AppFileInfo.newBuilder()
            .setFileName("save.dat")
            .setPathPrefixIndex(0)
            .build()

        val info = AppFileInfo(response)

        assertTrue(info.hasPathPrefixIndex)
        assertEquals(0, info.pathPrefixIndex)
    }
}
